RecoMuon/MuonSeedGenerator/python/ancientMuonSeed_cfi.py

```diff
@@ -1,23 +1,26 @@
 import FWCore.ParameterSet.Config as cms
 
-from RecoMuon.MuonSeedGenerator.ptSeedParameterization_cfi import *
-from RecoMuon.MuonSeedGenerator.MuonSeedPtScale_cfi import *
+from RecoMuon.MuonSeedGenerator.ptSeedParameterization_cfi import ptSeedParameterization
+from RecoMuon.MuonSeedGenerator.MuonSeedPtScale_cfi import dphiScale
 
 import RecoMuon.MuonSeedGenerator.produceMuons_cfi as _mod
 ancientMuonSeed = _mod.produceMuons.clone(
                                  ptSeedParameterization,
                                  dphiScale,
-                                 beamSpotTag = cms.InputTag("offlineBeamSpot"),
-                                 scaleDT = cms.bool(True),
-                                 CSCRecSegmentLabel = cms.InputTag("cscSegments"),
-                                 DTRecSegmentLabel = cms.InputTag("dt4DSegments"),
-                                 ME0RecSegmentLabel = cms.InputTag("me0Segments"),
+                                 beamSpotTag = "offlineBeamSpot",
```

Comment:
Once you have defined all these parameters in the `fillDescriptions` method of MuonSeedGenerator, you can proceed as usual and remove their redefinition from this cloned config. (Only let the two PSet's if you follow my suggestion above and allow them with `setUnknown()`

Reply:
@perrotta Thanks a lot for your comments. I've followed your suggestion to my codes and tested them in my lxplus. But a similar error occurred (like [PR33354](https://github.com/cms-sw/cmssw/pull/33354#issuecomment-820533852):[sol1](https://github.com/cms-sw/cmssw/pull/33354#issuecomment-824309911) 
[sol2](https://github.com/cms-sw/cmssw/pull/33354#issuecomment-824309911) ). The exact error message was : 
```
----- Begin Fatal Exception 05-May-2021 09:40:15 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=MuonSeedGenerator label='ancientMuonSeed'
Exception Message:
Illegal parameters found in configuration.  The parameters are named:
 'CSC_01'
 'CSC_01_1_scale'
 'CSC_02'
...
 'SME_22_0_scale'
 'SME_31'
 'SME_32'
 'SME_41'
 'SME_42'
You could be trying to use parameter names that are not
allowed for this plugin or they could be misspelled.
----- End Fatal Exception -------------------------------------------------
```

RecoMuon/MuonSeedGenerator/plugins/MuonSeedGenerator.cc is updated [as you commented above](https://github.com/cms-sw/cmssw/pull/33563#discussion_r625526434).
```
void MuonSeedGenerator::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
  edm::ParameterSetDescription desc;
  {
      edm::ParameterSetDescription psd0;
      psd0.setUnknown();
      desc.add<edm::ParameterSetDescription>("ptSeedParameterization", psd0);
   }
   {
      edm::ParameterSetDescription psd0;
      psd0.setUnknown();
      desc.add<edm::ParameterSetDescription>("dphiScale", psd0);
   }
  desc.add<edm::InputTag>("beamSpotTag", edm::InputTag("offlineBeamSpot"));
  desc.add<bool>("scaleDT", true);
  desc.add<edm::InputTag>("CSCRecSegmentLabel", edm::InputTag("cscSegments"));
  desc.add<edm::InputTag>("DTRecSegmentLabel", edm::InputTag("dt4DSegments"));
  desc.add<edm::InputTag>("ME0RecSegmentLabel", edm::InputTag("me0Segments"));
  desc.add<bool>("EnableDTMeasurement", true);
  desc.add<bool>("EnableCSCMeasurement", true);
  desc.add<bool>("EnableME0Measurement", false);
  desc.add<std::vector<double>>("crackEtas", {0.2, 1.6, 1.7});
  desc.add<double>("crackWindow", 0.04);
  desc.add<double>("deltaPhiSearchWindow", 0.25);
  desc.add<double>("deltaEtaSearchWindow", 0.2);
  desc.add<double>("deltaEtaCrackSearchWindow", 0.25);
  descriptions.add("muonSeedGenerator", desc);
```

RecoMuon/MuonSeedGenerator/python/ancientMuonSeed_cfi.py is updated:
```
import FWCore.ParameterSet.Config as cms
from RecoMuon.MuonSeedGenerator.ptSeedParameterization_cfi import ptSeedParameterization
from RecoMuon.MuonSeedGenerator.MuonSeedPtScale_cfi import dphiScale
import RecoMuon.MuonSeedGenerator.muonSeedGenerator_cfi as _mod
ancientMuonSeed = _mod.muonSeedGenerator.clone(
                                ptSeedParameterization,
                                dphiScale
                                )
```
Please let me know if there's something wrong with my codes.

Reply:
Ok, don't mind @jeongeun .
I was convinced that the `setUnknown()` would have worked, but I didn't try: now I did, and I'm getting the very same errors as you. Actually, I also tried with 
```PYTHON
    psd0.addWildcard<std::vector<double>>("*");
```
which seemed to me more in line with the configuration content, but also that did not work,

The problem comes from the fact that the configuration in RecoMuon/MuonSeedGenerator/python/ancientMuonSeed_cfi.py apparently imports the parameters included in the ptSeedParameterization_cfi.py and MuonSeedPtScale_cfi.py themselves, and not the whole PSets (see dump below). As such, the fillDescription method includes the empty PSets that should have been taken from those two files instead, and the validation fails.

```PYTHON
process.ancientMuonSeed = cms.EDProducer("MuonSeedGenerator",
    CSCRecSegmentLabel = cms.InputTag("cscSegments"),
    CSC_01 = cms.vdouble(
        0.166, 0.0, 0.0, 0.031, 0.0, 
        0.0
    ),
    CSC_01_1_scale = cms.vdouble(-1.915329, 0.0),
    CSC_02 = cms.vdouble(
        0.612, -0.207, -0.0, 0.067, -0.001, 
        0.0
    ),
    CSC_03 = cms.vdouble(
        0.787, -0.338, 0.029, 0.101, -0.008, 
        0.0
    ),
    CSC_12 = cms.vdouble(
        -0.161, 0.254, -0.047, 0.042, -0.007, 
        0.0
    ),
    CSC_12_1_scale = cms.vdouble(-6.434242, 0.0),
    CSC_12_2_scale = cms.vdouble(-1.63622, 0.0),
    CSC_12_3_scale = cms.vdouble(-1.63622, 0.0),
    CSC_13 = cms.vdouble(
        0.901, -1.302, 0.533, 0.045, 0.005, 
        0.0
    ),
    CSC_13_2_scale = cms.vdouble(-6.077936, 0.0),
    CSC_13_3_scale = cms.vdouble(-1.701268, 0.0),
    CSC_14 = cms.vdouble(
        0.606, -0.181, -0.002, 0.111, -0.003, 
        0.0
    ),
    CSC_14_3_scale = cms.vdouble(-1.969563, 0.0),
    CSC_23 = cms.vdouble(
        -0.081, 0.113, -0.029, 0.015, 0.008, 
        0.0
    ),
    CSC_23_1_scale = cms.vdouble(-19.084285, 0.0),
    CSC_23_2_scale = cms.vdouble(-6.079917, 0.0),
    CSC_24 = cms.vdouble(
        0.004, 0.021, -0.002, 0.053, 0.0, 
        0.0
    ),
    CSC_24_1_scale = cms.vdouble(-6.055701, 0.0),
    CSC_34 = cms.vdouble(
        0.062, -0.067, 0.019, 0.021, 0.003, 
        0.0
    ),
    CSC_34_1_scale = cms.vdouble(-11.520507, 0.0),
    DTRecSegmentLabel = cms.InputTag("dt4DSegments"),
    DT_12 = cms.vdouble(
        0.183, 0.054, -0.087, 0.028, 0.002, 
        0.0
    ),
    DT_12_1_scale = cms.vdouble(-3.692398, 0.0),
    DT_12_2_scale = cms.vdouble(-3.518165, 0.0),
    DT_13 = cms.vdouble(
        0.315, 0.068, -0.127, 0.051, -0.002, 
        0.0
    ),
    DT_13_1_scale = cms.vdouble(-4.520923, 0.0),
    DT_13_2_scale = cms.vdouble(-4.257687, 0.0),
    DT_14 = cms.vdouble(
        0.359, 0.052, -0.107, 0.072, -0.004, 
        0.0
    ),
    DT_14_1_scale = cms.vdouble(-5.644816, 0.0),
    DT_14_2_scale = cms.vdouble(-4.808546, 0.0),
    DT_23 = cms.vdouble(
        0.13, 0.023, -0.057, 0.028, 0.004, 
        0.0
    ),
    DT_23_1_scale = cms.vdouble(-5.320346, 0.0),
    DT_23_2_scale = cms.vdouble(-5.117625, 0.0),
    DT_24 = cms.vdouble(
        0.176, 0.014, -0.051, 0.051, 0.003, 
        0.0
    ),
    DT_24_1_scale = cms.vdouble(-7.490909, 0.0),
    DT_24_2_scale = cms.vdouble(-6.63094, 0.0),
    DT_34 = cms.vdouble(
        0.044, 0.004, -0.013, 0.029, 0.003, 
        0.0
    ),
    DT_34_1_scale = cms.vdouble(-13.783765, 0.0),
    DT_34_2_scale = cms.vdouble(-11.901897, 0.0),
    EnableCSCMeasurement = cms.bool(True),
    EnableDTMeasurement = cms.bool(True),
    EnableME0Measurement = cms.bool(False),
    ME0RecSegmentLabel = cms.InputTag("me0Segments"),
    OL_1213 = cms.vdouble(
        0.96, -0.737, 0.0, 0.052, 0.0, 
        0.0
    ),
    OL_1213_0_scale = cms.vdouble(-4.488158, 0.0),
    OL_1222 = cms.vdouble(
        0.848, -0.591, 0.0, 0.062, 0.0, 
        0.0
    ),
    OL_1222_0_scale = cms.vdouble(-5.810449, 0.0),
    OL_1232 = cms.vdouble(
        0.184, 0.0, 0.0, 0.066, 0.0, 
        0.0
    ),
    OL_1232_0_scale = cms.vdouble(-5.964634, 0.0),
    OL_2213 = cms.vdouble(
        0.117, 0.0, 0.0, 0.044, 0.0, 
        0.0
    ),
    OL_2213_0_scale = cms.vdouble(-7.239789, 0.0),
    OL_2222 = cms.vdouble(
        0.107, 0.0, 0.0, 0.04, 0.0, 
        0.0
    ),
    OL_2222_0_scale = cms.vdouble(-7.667231, 0.0),
    SMB_10 = cms.vdouble(
        1.387, -0.038, 0.0, 0.19, 0.0, 
        0.0
    ),
    SMB_10_0_scale = cms.vdouble(2.448566, 0.0),
    SMB_11 = cms.vdouble(
        1.247, 0.72, -0.802, 0.229, -0.075, 
        0.0
    ),
    SMB_11_0_scale = cms.vdouble(2.56363, 0.0),
    SMB_12 = cms.vdouble(
        2.128, -0.956, 0.0, 0.199, 0.0, 
        0.0
    ),
    SMB_12_0_scale = cms.vdouble(2.283221, 0.0),
    SMB_20 = cms.vdouble(
        1.011, -0.052, 0.0, 0.188, 0.0, 
        0.0
    ),
    SMB_20_0_scale = cms.vdouble(1.486168, 0.0),
    SMB_21 = cms.vdouble(
        1.043, -0.124, 0.0, 0.183, 0.0, 
        0.0
    ),
    SMB_21_0_scale = cms.vdouble(1.58384, 0.0),
    SMB_22 = cms.vdouble(
        1.474, -0.758, 0.0, 0.185, 0.0, 
        0.0
    ),
    SMB_22_0_scale = cms.vdouble(1.346681, 0.0),
    SMB_30 = cms.vdouble(
        0.505, -0.022, 0.0, 0.215, 0.0, 
        0.0
    ),
    SMB_30_0_scale = cms.vdouble(-3.629838, 0.0),
    SMB_31 = cms.vdouble(
        0.549, -0.145, 0.0, 0.207, 0.0, 
        0.0
    ),
    SMB_31_0_scale = cms.vdouble(-3.323768, 0.0),
    SMB_32 = cms.vdouble(
        0.67, -0.327, 0.0, 0.22, 0.0, 
        0.0
    ),
    SMB_32_0_scale = cms.vdouble(-3.054156, 0.0),
    SME_11 = cms.vdouble(
        3.295, -1.527, 0.112, 0.378, 0.02, 
        0.0
    ),
    SME_11_0_scale = cms.vdouble(1.325085, 0.0),
    SME_12 = cms.vdouble(
        0.102, 0.599, 0.0, 0.38, 0.0, 
        0.0
    ),
    SME_12_0_scale = cms.vdouble(2.279181, 0.0),
    SME_13 = cms.vdouble(
        -1.286, 1.711, 0.0, 0.356, 0.0, 
        0.0
    ),
    SME_13_0_scale = cms.vdouble(0.104905, 0.0),
    SME_21 = cms.vdouble(
        -0.529, 1.194, -0.358, 0.472, 0.086, 
        0.0
    ),
    SME_21_0_scale = cms.vdouble(-0.040862, 0.0),
    SME_22 = cms.vdouble(
        -1.207, 1.491, -0.251, 0.189, 0.243, 
        0.0
    ),
    SME_22_0_scale = cms.vdouble(-3.457901, 0.0),
    SME_31 = cms.vdouble(
        -1.594, 1.482, -0.317, 0.487, 0.097, 
        0.0
    ),
    SME_32 = cms.vdouble(
        -0.901, 1.333, -0.47, 0.41, 0.073, 
        0.0
    ),
    SME_41 = cms.vdouble(
        -0.003, 0.005, 0.005, 0.608, 0.076, 
        0.0
    ),
    SME_42 = cms.vdouble(
        -0.003, 0.005, 0.005, 0.608, 0.076, 
        0.0
    ),
    beamSpotTag = cms.InputTag("offlineBeamSpot"),
    crackEtas = cms.vdouble(0.2, 1.6, 1.7),
    crackWindow = cms.double(0.04),
    deltaEtaCrackSearchWindow = cms.double(0.25),
    deltaEtaSearchWindow = cms.double(0.2),
    deltaPhiSearchWindow = cms.double(0.25),
    dphiScale = cms.PSet(

    ),
    mightGet = cms.optional.untracked.vstring,
    ptSeedParameterization = cms.PSet(

    ),
    scaleDT = cms.bool(True)
)
```

Reply:
There is probably some python magic we can attempt to solve this issues (i.e. making sure that the `ptSeedParameterization` PSet from RecoMuon/MuonSeedGenerator/python/ptSeedParameterization_cfi.py gets included as a whole PSet instead of including all its parameters separately). I will give it a thought, but I think that this is starting to get a bit farther from the original intent of this pull request, and we could even stop updating here and accept this working version.

Reply:
Okay. Thanks @perrotta !
I only renamed the default cfi "produceMuons" to "muonSeedGenerator".